src/test/java/co/rsk/federate/btcreleaseclient/cache/PegoutSignedCacheImplTest.java

```diff
@@ -0,0 +1,113 @@
+    Field field = pegoutSignedCache.getClass().getDeclaredField("cache");
+    field.setAccessible(true);
+    field.set(pegoutSignedCache, cache);
+  }
+
+  @Test
+  void hasAlreadyBeenSigned_shouldReturnFalse_whenPegoutCreationRskTxHashIsNull() {
+    Keccak256 pegoutCreationRskTxHash = null;
+
+    boolean result = pegoutSignedCache.hasAlreadyBeenSigned(pegoutCreationRskTxHash);
+
+    assertFalse(result);
+  }
+
+  @Test
+  void hasAlreadyBeenSigned_shouldReturnFalse_whenPegoutCreationRskTxHashIsNotCached() {
+    boolean result = pegoutSignedCache.hasAlreadyBeenSigned(PEGOUT_CREATION_RSK_HASH);
+
+    assertFalse(result);
+  }
+
+  @Test
+  void hasAlreadyBeenSigned_shouldReturnFalse_whenCacheContainsExpiredTimestamp() {
+    Instant currentTimestamp = Instant.now();
+    Instant timestamp = currentTimestamp.minusMillis(Duration.ofMinutes(60).toMillis());
+    cache.put(PEGOUT_CREATION_RSK_HASH, timestamp);
+
+    boolean result = pegoutSignedCache.hasAlreadyBeenSigned(PEGOUT_CREATION_RSK_HASH);
+
+    assertFalse(result);
+  }
+
+  @Test
+  void hasAlreadyBeenSigned_shouldReturnTrue_whenCacheContainsNotExpiredTimestamp() {
+    Instant currentTimestamp = Instant.now();
+    Instant timestamp = currentTimestamp.minusMillis(Duration.ofMinutes(10).toMillis());
+    cache.put(PEGOUT_CREATION_RSK_HASH, timestamp);
+
+    boolean result = pegoutSignedCache.hasAlreadyBeenSigned(PEGOUT_CREATION_RSK_HASH);
+
+    assertTrue(result);
+  }
+
+  @Test
+  void putIfAbsent_shouldNotPutInCache_whenPegoutCreationRskTxHashIsNull() {
+    Keccak256 pegoutCreationRskTxHash = null;
+
+    pegoutSignedCache.putIfAbsent(pegoutCreationRskTxHash);
+
+    assertFalse(cache.containsKey(PEGOUT_CREATION_RSK_HASH));
+  }
+
+  @Test
+  void putIfAbsent_shouldPutInCache_whenPegoutCreationRskTxHashIsNotNull() {
+    pegoutSignedCache.putIfAbsent(PEGOUT_CREATION_RSK_HASH);
+
+    assertTrue(cache.containsKey(PEGOUT_CREATION_RSK_HASH));
+  }
+
+  @Test
+  void putIfAbsent_shouldPutInCacheBoth_whenPegoutCreationRskTxHashAreNotSame() {
+    // first insert
+    pegoutSignedCache.putIfAbsent(PEGOUT_CREATION_RSK_HASH);
+    Instant pegoutCreationRskTxHashTimestamp = cache.get(PEGOUT_CREATION_RSK_HASH);
+    // second insert
+    Keccak256 otherPegoutCreationRskTxHash = TestUtils.createHash(2);
+    pegoutSignedCache.putIfAbsent(otherPegoutCreationRskTxHash);
+    Instant otherPegoutCreationRskTxHashTimestamp = cache.get(otherPegoutCreationRskTxHash);
+
+    assertNotSame(pegoutCreationRskTxHashTimestamp, otherPegoutCreationRskTxHashTimestamp);
+  }
+
+  @Test
+  void putIfAbsent_shouldPutInCacheOnce_whenPegoutCreationRskTxHashIsTheSame() {
+    // first insert
+    pegoutSignedCache.putIfAbsent(PEGOUT_CREATION_RSK_HASH);
+    Instant pegoutCreationRskTxHashTimestamp = cache.get(PEGOUT_CREATION_RSK_HASH);
+    // second insert
+    pegoutSignedCache.putIfAbsent(PEGOUT_CREATION_RSK_HASH);
+    Instant otherPegoutCreationRskTxHashTimestamp = cache.get(PEGOUT_CREATION_RSK_HASH);
```

Comment:
`Instant pegoutCreationRskTxHashTimestamp2 = cache.get(PEGOUT_CREATION_RSK_HASH);`

the "other" at the beginning makes me think that you are talking about another pegout and not about another timestamp (overall because above is being used that way)

Reply:
a good idea could be checking for cache size, as you did [here](https://github.com/rsksmart/powpeg-node/blob/25acf0f329e2279715b9d44da1d255d3f9cbc1cb/src/test/java/co/rsk/federate/btcreleaseclient/cache/PegoutSignedCacheImplTest.java#L86)

Reply:
Something similar is what I originally had, will update to this.